src/realm/object-store/c_api/sync.cpp

```diff
@@ -413,6 +413,155 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_latest_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    return new realm_flx_sync_subscription_set_t((*realm)->get_latest_subscription_set());
+}
+
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_active_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    return new realm_flx_sync_subscription_set_t((*realm)->get_active_subscription_set());
+}
+
+RLM_API realm_flx_sync_mutable_subscription_set_t*
+realm_sync_get_mutable_subscription_set(const realm_flx_sync_subscription_set_t* subscription_set)
+{
+    if (subscription_set == nullptr)
+        return nullptr;
+    return new realm_flx_sync_mutable_subscription_set_t(subscription_set->make_mutable_copy());
+}
+
+RLM_API realm_flx_sync_subscription_set_state_e realm_sync_get_state_change_notification(
+    const realm_flx_sync_subscription_set_t* subscription_set, realm_flx_sync_subscription_set_state_e notify_when)
```

Comment:
@jbreams  this has become a blocking interface in the c-api, the original design returns a future type, which it is no usable for obvious reasons in the c-api. I wonder if we may want to use get_async(...) here, even though it is not 100% clear what the sign of the future should look like. Probably, we should just pass the state when it is ready.

Reply:
yeah, I don't think this needs to be blocking. You can do get_async and the user-supplied callback will be called on the sync client's worker thread. Do we want to let users pass in a `user_data` void* as a parameter here?

Reply:
API now is async